src/anchorfm-pupeteer/index.js

```diff
@@ -72,10 +72,12 @@ async function selectMonthInDatePicker(page, navigationPromise, month) {
 
 async function selectDayInDatePicker(page, navigationPromise, day) {
   const dayWithRemovedZeroPad = parseInt(day, 10);
-  const tdDay = await page.$(
-    `tbody > tr > td[data-value="${dayWithRemovedZeroPad}"][class*="rdtDay"]`
+  const tdDay = await page.$x(
```

Comment:
Explanation:

Instead of using the CSS selector `tbody > tr > td[data-value="${dayWithRemovedZeroPad}"][class*="rdtDay"]:not([class*="rdtOld"]:not([class*="rtdNew"])`, we use an XPath expression that selects the same element:
```
//tbody/tr/td[contains(concat(' ', normalize-space(@class), ' '), ' rdtDay ')]
  [not(contains(concat(' ', normalize-space(@class), ' '), ' rdtOld ')) and not(contains(concat(' ', normalize-space(@class), ' '), ' rdtNew '))]
  [@data-value="${dayWithRemovedZeroPad}"]`
```
The XPath expression uses the contains() function to match the rdtDay, rdtOld, and rdtNew classes, and the not() function to exclude elements with those classes.
To click on the selected element, we use the `page.$x()` method, which returns an array of elements that match the XPath expression. We access the first element using `tdDay[0]`.
We then click on the element and wait for the navigation to complete, just like in the original function.

Reply:
Update:

There was a puppeteer bug related to using `:not`, and it seems to be fixed in latest puppeteer 19.7.5.

Please update puppeteer, use my fixed selector, and test :)